mat/cholesky.go

```diff
@@ -287,11 +287,24 @@ func (c *Cholesky) ToSym(dst *SymDense) *SymDense {
 	}
 	n := c.chol.mat.N
 	if dst == nil {
-		dst = NewSymDense(n, make([]float64, n*n))
+		dst = NewSymDense(n, nil)
 	} else {
 		dst.reuseAs(n)
 	}
-	dst.SymOuterK(1, c.chol.T())
+	// Copy c.chol into a TriDense U with dst's backing slice.
```

Comment:
I'm not very happy about this part, so if you have any suggestion how to do this better, I'll appreciate it.

Reply:
It is a bit funky. It took me a while to figure out what is going on, though I don't see a better way to do it. Maybe just an additional comment saying that work on `u` is reflected into `dst` and do the work on `u`, like so:
```
	// Copy c.chol into a TriDense U with dst's backing slice.
	// Operations on u are reflected in dst.
	u := *c.chol
	u.mat.Data = dst.mat.Data
	u.Copy(c.chol)
	// Compute the product U^T*U using the algorithm from LAPACK/TESTING/LIN/dpot01.f
	bi := blas64.Implementation()
	a := u.mat.Data
	lda := u.mat.Stride
	for k := n - 1; k >= 0; k-- {
		a[k*lda+k] = bi.Ddot(k+1, a[k:], lda, a[k:], lda)
		if k > 0 {
			bi.Dtrmv(blas.Upper, blas.Trans, blas.NonUnit, k, a, lda, a[k:], lda)
		}
	}
	return dst
```

Reply:
Funky is the word :-) I added the comment and do the work on u, thanks.